backend/services/openai.go

```diff
@@ -16,11 +16,17 @@ var OPEN_AI_KEY string
 
 func Init() {
 	OPEN_AI_KEY := os.Getenv("OPEN_AI_KEY")
-	OpenAIclient = openai.NewClient(OPEN_AI_KEY)
+	OPEN_AI_SERVER_URL := os.Getenv("OPEN_AI_SERVER_URL")
+	cfg := openai.DefaultConfig(OPEN_AI_KEY)
+	cfg.BaseURL = OPEN_AI_SERVER_URL
+	OpenAIclient = openai.NewClientWithConfig(cfg)
 
 	if OPEN_AI_KEY == "" {
 		log.Fatal("OPEN_AI_KEY is not set")
 	}
+	if OPEN_AI_SERVER_URL == "" {
```

Comment:
This can be removed after moving `OpenAIServerURL` to the config.go since it already has this logic build in.

Reply:
For more context: we use `RequiredIfNoDef` param so it fails if there are no value by default.

https://pkg.go.dev/github.com/caarlos0/env/v10#readme-making-all-fields-to-required